src/subscriptionModules/eventScannerBased.ts

```diff
@@ -0,0 +1,252 @@
+    }
+
+    private _requestNewScan = async (): Promise<void> => {
+      if(this.isScanOngoing){
+        this.isNewScanRequired = true
+        this.logger.info(`new scan queued...`)
+      }else{
+        try {
+          await this._scanForTransferEvents()
+        } catch (error) {
+          this.logger.error(`last SCAN had an issue !: ${JSON.stringify(error)}`)
+          this.isNewScanRequired = true
+        } finally {
+          this.isScanOngoing = false
+        }
+
+        if(this.isNewScanRequired){
+          this._requestNewScan()
+        }
+      } 
+    }
+
+    private _scanForTransferEvents = async (): Promise<void> => {
+      this.isScanOngoing = true
+      this.isNewScanRequired = false
+
+      const currentBlockNumber = (await this.api.rpc.chain.getHeader()).number.unwrap().toNumber()
+      const lastCheckedBlock = await this._getLastCheckedBlock()
+      let result = true // important to decide whether to continue the loop or not
+      this.logger.info(`\n*****\nStarting a new SCAN...\nStarting Block: ${lastCheckedBlock+1}\nEnding Block: ${currentBlockNumber}\n*****`)
+      for (let blockNumber = lastCheckedBlock + 1; blockNumber <= currentBlockNumber; result == true ? blockNumber++ : {} ){
+
+        // important to decide whether to continue the loop or not
+        result = true
+
+        const blockHash = await this.api.rpc.chain.getBlockHash(blockNumber)
+        const block = await this.api.rpc.chain.getBlock(blockHash)
+        const allRecords = await this.api.query.system.events.at(blockHash);
+
+        for (const [index, { hash }] of block.block.extrinsics.entries()) {
+          for (const {event} of allRecords
+            .filter(({ phase,event }) => 
+              phase.isApplyExtrinsic &&
+              phase.asApplyExtrinsic.eq(index) && 
+              isBalanceTransferEvent(event,this.api)
+            )) {
+
+              let retriesLeft = 3
+              do {
+                result = await this._balanceTransferHandler(event, hash)
+                if(!result){
+                  retriesLeft--
+                  this.logger.warn(`New retry at block ${blockNumber} !!`)
+                  await delay(5000)
+                }
+                
+              } while (!result && retriesLeft > 0);
+          }
+        }
+
+        if(result){
+          this.logger.debug(`Updated lastCheckedBlock to ${blockNumber} !!`)
+          result = await this._updateLastCheckedBlock(blockNumber)
+        }
+
+        if(!result){
+          // better to risk to send them all again (block) than loose one of them
+          this.logger.warn(`\n*****\nSCAN had an issue at block ${blockNumber}, exiting...\n*****`)
+          throw `SCAN ERROR at block ${blockNumber}`
+        }
+      }
+      
+      if(result){
+        this.logger.info(`\n*****\nSCAN completed at block ${await this._getLastCheckedBlock()}\n*****`)
+      }else{
```

Comment:
I don't believe this code path is ever reachable, therefore not needed. Either you throw at line 150 or result is still true at this point.

Reply:
True, I wanted to be really sure =)